.github/workflows/other.yml

```diff
@@ -0,0 +1,138 @@
+name: Build and test (other)
```

Comment:
why is separating necessary? To give more space to those that do not need dune cache?

Reply:
There is a built-in calculation in the 'setup-ocaml' github action that trims the dune cache size to 5GiB/jobs-in-workflow. So by ensuring that jobs-in-workflow=1 we are entirely in control of the dune cache size, and not the github action (so we can tune it up and down as we need to).

The commit message explains it:
```
    ci: separate workflows

    The dune cache inside 'setup-ocaml' is trimmed to 5GB/jobcount,
    which is currently about 10.
    But not all jobs here use the dune cache, so separate the workflows.
    (Also the dune cache will get compressed afterwards).

    The SDK build does use the dune cache, but eventually it might become smaller,
    e.g. if we change it to install just the dependencies needed for SDK gen,
    and not everything. It does have a lot of unrelated jobs though.
```

We may also be able to trim down the number of `opam` packages we need to install for the SDK job eventually, so a separate (smaller) dune cache will be better for it.